ethsigner/src/main/java/tech/pegasys/ethsigner/requesthandler/sendtransaction/TransactionTransmitter.java

```diff
@@ -0,0 +1,145 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.requesthandler.sendtransaction;
+
+import static java.util.Collections.singletonList;
+
+import tech.pegasys.ethsigner.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.jsonrpc.response.JsonRpcError;
+import tech.pegasys.ethsigner.jsonrpc.response.JsonRpcErrorResponse;
+import tech.pegasys.ethsigner.requesthandler.JsonRpcBody;
+import tech.pegasys.ethsigner.signing.TransactionSigner;
+
+import java.io.IOException;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.json.Json;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.web3j.crypto.RawTransaction;
+
+public class TransactionTransmitter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TransactionTransmitter.class);
+  private static final String JSON_RPC_VERSION = "2.0";
+  private static final String JSON_RPC_METHOD = "eth_sendRawTransaction";
+
+  private final HttpClient ethNodeClient;
+  private final TransactionSigner signer;
+  private final TransactionInformation transactionInfo;
+  private final RetryMechanism retryMechanism;
+
+  public TransactionTransmitter(
+      final HttpClient ethNodeClient,
+      final TransactionInformation transactionInfo,
+      final TransactionSigner signer,
+      final RetryMechanism retryMechanism) {
+    this.ethNodeClient = ethNodeClient;
+    this.transactionInfo = transactionInfo;
+    this.signer = signer;
+    this.retryMechanism = retryMechanism;
+  }
+
+  public void send() {
+    final JsonRpcBody body = getBody();
+    if (body.hasError()) {
+      // ????? DO SOMETHING ????
+    } else {
+      sendTransaction(body.body());
+    }
+  }
+
+  private JsonRpcBody getBody() {
+    // This assumes the parameters have already been validated and are correct for the unlocked
+    // account.
+    final String signedTransactionHexString;
+    try {
+      final RawTransaction rawTransaction = transactionInfo.getRawTransactionBuilder().build();
+      signedTransactionHexString = signer.signTransaction(rawTransaction);
+    } catch (final IllegalArgumentException e) {
+      LOG.debug("Bad input value from request: {}", "UNKNOWN", e);
+      return createJsonRpcBodyFrom(JsonRpcError.INVALID_PARAMS);
+    } catch (final Throwable e) {
+      LOG.debug("Unhandled error processing request: {}", "UNKNOWN", e);
+      return createJsonRpcBodyFrom(JsonRpcError.INTERNAL_ERROR);
```

Comment:
Output string doesn't need variable substitution i.e. `"Unhandled error processing request: UNKNOWN"` ...although I'd expect a better course would be to output the request rather then a string constant

Reply:
Done.